File: src/dataset/Java/space-age_exercism-uzilan/gemini/GeminiFlash_SpaceAge_v1.java
class SpaceAge {

    private final double seconds;
    private static final double EARTH_ORBITAL_PERIOD = 31557600.0;
    private static final double SECONDS_PER_EARTH_YEAR = EARTH_ORBITAL_PERIOD;

    private enum OrbitalPeriod {
        MERCURY(0.2408467),
        VENUS(0.61519726),
        MARS(1.8808158),
        JUPITER(11.862615),
        SATURN(29.447498),
        URANUS(84.016846),
        NEPTUNE(164.79132);

        final double orbitalPeriodInEarthYears;

        OrbitalPeriod(double orbitalPeriodInEarthYears) {
            this.orbitalPeriodInEarthYears = orbitalPeriodInEarthYears;
        }
    }

    SpaceAge(double seconds) {
        this.seconds = seconds;
    }

    private double calculateAge(double orbitalPeriodInEarthYears) {
        return seconds / (SECONDS_PER_EARTH_YEAR * orbitalPeriodInEarthYears);
    }

    double onEarth() {
        return seconds / SECONDS_PER_EARTH_YEAR;
    }

    double onMercury() {
        return calculateAge(OrbitalPeriod.MERCURY.orbitalPeriodInEarthYears);
    }

    double onVenus() {
        return calculateAge(OrbitalPeriod.VENUS.orbitalPeriodInEarthYears);
    }

    double onMars() {
        return calculateAge(OrbitalPeriod.MARS.orbitalPeriodInEarthYears);
    }

    double onJupiter() {
        return calculateAge(OrbitalPeriod.JUPITER.orbitalPeriodInEarthYears);
    }

    double onSaturn() {
        return calculateAge(OrbitalPeriod.SATURN.orbitalPeriodInEarthYears);
    }

    double onUranus() {
        return calculateAge(OrbitalPeriod.URANUS.orbitalPeriodInEarthYears);
    }

    double onNeptune() {
        return calculateAge(OrbitalPeriod.NEPTUNE.orbitalPeriodInEarthYears);
    }
}

File: src/dataset/typescript/space-age_exercism-wackerow/gemini/GeminiFlash_space-age_v1.ts
class SpaceAge {
  private static readonly earthYearSeconds = 31557600;

  private static readonly orbitalPeriods: { [planet: string]: number } = {
    mercury: 0.2408467,
    venus: 0.61519726,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132,
  };

  private readonly secondsInEarthYears: number;

  constructor (public seconds: number) {
    this.secondsInEarthYears = seconds / SpaceAge.earthYearSeconds;
  }

  onEarth(): number {
    return this.roundToTwo(this.secondsInEarthYears);
  }

  onMercury(): number {
    return this.roundToTwo(this.secondsInEarthYears / SpaceAge.orbitalPeriods.mercury);
  }
  onVenus(): number {
    return this.roundToTwo(this.secondsInEarthYears / SpaceAge.orbitalPeriods.venus);
  }
  onMars(): number {
    return this.roundToTwo(this.secondsInEarthYears / SpaceAge.orbitalPeriods.mars);
  }
  onJupiter(): number {
    return this.roundToTwo(this.secondsInEarthYears / SpaceAge.orbitalPeriods.jupiter);
  }
  onSaturn(): number {
    return this.roundToTwo(this.secondsInEarthYears / SpaceAge.orbitalPeriods.saturn);
  }
  onUranus(): number {
    return this.roundToTwo(this.secondsInEarthYears / SpaceAge.orbitalPeriods.uranus);
  }
  onNeptune(): number {
    return this.roundToTwo(this.secondsInEarthYears / SpaceAge.orbitalPeriods.neptune);
  }

  private roundToTwo(value: number): number {
    return Math.round(value * 100) / 100;
  }
}

export default SpaceAge;